src/app/filecoin-plus/allocators/components/AllocatorsTableWithFilters.tsx

```diff
@@ -22,30 +30,55 @@ type AllocatorsTableWithFiltersProps = {
 export function AllocatorsTableWithFilters({
   data,
 }: AllocatorsTableWithFiltersProps) {
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([])
+
+  const [sorting, setSorting] = useState<SortingState>([
+    { id: 'name', desc: false },
+  ])
+
   const table = useReactTable({
     data,
     columns: allocatorsTableColumnsData,
+    state: { columnFilters, sorting },
+    onSortingChange: setSorting,
+    onColumnFiltersChange: setColumnFilters,
     getCoreRowModel: getCoreRowModel(),
     getSortedRowModel: getSortedRowModel(),
     getFilteredRowModel: getFilteredRowModel(),
-    initialState: {
-      sorting: [{ id: 'name', desc: false }],
-    },
-    globalFilterFn: 'includesString',
   })
-  const { getState, setGlobalFilter, getRowModel, getHeaderGroups } = table
+  const { getRowModel, getHeaderGroups } = table
 
-  const searchQuery = getState().globalFilter || ''
   const rowModel = getRowModel()
   const headerGroups = getHeaderGroups()
 
+  const nameColumn = getColumnOrThrow(table, 'name')
+  const locationColumn = getColumnOrThrow(table, 'location')
+  const typeColumn = getColumnOrThrow(table, 'metapathway_type')
+
+  const locationOptions = createUniqueOptionsFromData(data, 'location')
+  const typeOptions = createUniqueOptionsFromData(data, 'metapathway_type')
+
   const hasSearchResults = Boolean(rowModel.rows.length)
 
   return (
     <>
-      <div className="mb-6 max-w-readable">
-        <SearchInput query={searchQuery} onChange={setGlobalFilter} />
-      </div>
+      <AllocatorsTableFiltersLayout
+        nameSearch={<SearchFilter column={nameColumn} />}
+        typeFilter={
+          <SelectFilter
+            column={typeColumn}
+            options={typeOptions}
+            resetOptionName="All Types"
+          />
+        }
+        locationFilter={
+          <SelectFilter
+            column={locationColumn}
+            options={locationOptions}
+            resetOptionName="All Regions"
+          />
+        }
+      />
```

Comment:
From [>>](https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/754/files#r1819025319)

```suggestion
          <AllocatorsTableFiltersLayout data={data} table={table} />
```

Reply:
I reworked it as per your suggestion